src/main/java/appeng/core/sync/packets/PacketMEInventoryUpdate.java

```diff
@@ -61,162 +58,169 @@
+			final GZIPInputStream gzReader = new GZIPInputStream( new InputStream()
 			{
-				if( stream.readableBytes() <= 0 )
+				@Override
+				public int read() throws IOException
 				{
-					return -1;
+					if( buf.readableBytes() <= 0 )
+					{
+						return -1;
+					}
+
+					return buf.readByte() & STREAM_MASK;
 				}
+			} );
 
-				return stream.readByte() & STREAM_MASK;
-			}
-		} );
+			final ByteBuf uncompressed = Unpooled.buffer( buf.readableBytes() );
+			final byte[] tmp = new byte[TEMP_BUFFER_SIZE];
 
-		final ByteBuf uncompressed = Unpooled.buffer( stream.readableBytes() );
-		final byte[] tmp = new byte[TEMP_BUFFER_SIZE];
-		while( gzReader.available() != 0 )
-		{
-			final int bytes = gzReader.read( tmp );
-			if( bytes > 0 )
+			while( gzReader.available() != 0 )
 			{
-				uncompressed.writeBytes( tmp, 0, bytes );
+				final int bytes = gzReader.read( tmp );
+				if( bytes > 0 )
+				{
+					uncompressed.writeBytes( tmp, 0, bytes );
+				}
 			}
-		}
-		gzReader.close();
 
-		// int uncompressedBytes = uncompressed.readableBytes();
-		// AELog.info( "Receiver: " + originalBytes + " -> " + uncompressedBytes );
+			gzReader.close();
 
-		while( uncompressed.readableBytes() > 0 )
+			while( uncompressed.readableBytes() > 0 )
+			{
+				this.list.add( AEItemStack.loadItemStackFromPacket( uncompressed ) );
+			}
+		}
+		catch( final IOException e )
 		{
-			this.list.add( AEItemStack.loadItemStackFromPacket( uncompressed ) );
 		}
 
 		this.empty = this.list.isEmpty();
 	}
 
-	// api
-	public PacketMEInventoryUpdate() throws IOException
+	@Override
+	public void toBytes( final ByteBuf buf )
 	{
-		this( (byte) 0 );
-	}
+		buf.writeByte( this.ref );
 
-	// api
-	public PacketMEInventoryUpdate( final byte ref ) throws IOException
-	{
-		this.ref = ref;
-		this.data = Unpooled.buffer( OPERATION_BYTE_LIMIT );
-		this.data.writeInt( this.getPacketID() );
-		this.data.writeByte( this.ref );
+		OutputStream compressFrame = null;
 
-		this.compressFrame = new GZIPOutputStream( new OutputStream()
+		try
 		{
-			@Override
-			public void write( final int value ) throws IOException
+			compressFrame = new GZIPOutputStream( new OutputStream()
 			{
-				PacketMEInventoryUpdate.this.data.writeByte( value );
-			}
-		} );
+				@Override
+				public void write( final int value ) throws IOException
+				{
+					buf.writeByte( value );
+				}
+			} );
+
+			for( final IAEItemStack iaeItemStack : this.list )
+			{
+				final ByteBuf tmp = Unpooled.buffer( OPERATION_BYTE_LIMIT );
 
-		this.list = null;
+				iaeItemStack.writeToPacket( tmp );
+
+				if( this.writtenBytes + tmp.readableBytes() > UNCOMPRESSED_PACKET_BYTE_LIMIT )
+				{
+					throw new BufferOverflowException();
+				}
+				else
+				{
+					this.writtenBytes += tmp.readableBytes();
+					compressFrame.write( tmp.array(), 0, tmp.readableBytes() );
+					this.empty = false;
+				}
+
+			}
+			compressFrame.flush();
+		}
+		catch( final IOException e1 )
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)